git_diff_branch...HEAD main  # at /cygdrive/e/Note/Tool/llvm started
diff --git a/llvm/CMakeLists.txt b/llvm/CMakeLists.txt
index f0e4f5d7d6f6..de099ca4ace6 100644
--- a/llvm/CMakeLists.txt
+++ b/llvm/CMakeLists.txt
@@ -933,6 +933,14 @@ if(LLVM_BUILD_LLVM_DYLIB_VIS)
   set(LLVM_BUILD_LLVM_DYLIB ON)
 endif()
 
+if((WIN32 OR CYGWIN) AND (NOT LLVM_ENABLE_PIC))
+	message(FATAL_ERROR "on Windows, LLVM_ENABLE_PIC should not be turned off.")
+endif()
+
+if ((LLVM_BUILD_LLVM_DYLIB AND NOT LLVM_LINK_LLVM_DYLIB) OR (NOT LLVM_BUILD_LLVM_DYLIB AND LLVM_LINK_LLVM_DYLIB))
+	message(FATAL_ERROR "LLVM_BUILD_LLVM_DYLIB and LLVM_LINK_LLVM_DYLIB should be the same.")
+endif()
+
 if (LLVM_LINK_LLVM_DYLIB AND BUILD_SHARED_LIBS)
   message(FATAL_ERROR "Cannot enable BUILD_SHARED_LIBS with LLVM_LINK_LLVM_DYLIB.  We recommend disabling BUILD_SHARED_LIBS.")
 endif()
